src/Symfony/Bundle/FrameworkBundle/Tests/Translation/TranslatorTest.php

```diff
@@ -91,6 +93,27 @@ public function testTransWithCaching()
         $this->assertEquals('other choice 1 (PT-BR)', $translator->transChoice('other choice', 1));
         $this->assertEquals('foobarbaz (fr.UTF-8)', $translator->trans('foobarbaz'));
         $this->assertEquals('foobarbax (sr@latin)', $translator->trans('foobarbax'));
+
+        // refresh cache again when resource file resources file change
+        $resource = $this->getMock('Symfony\Component\Config\Resource\ResourceInterface');
+        $resource
+            ->expects($this->at(0))
+            ->method('isFresh')
+            ->will($this->returnValue(false))
+        ;
+        $catalogue = $this->getCatalogue('fr', array('foo' => 'foo fresh'));
+        $catalogue->addResource($resource);
```

Comment:
@xabbuh haven't looked at it in detail, but I think this resource mock has no effect. Instead the resource already serialized in the cache will be loaded (and still be fresh)?